ThirdParty/Deferred/Deferred/ReadWriteLock.swift

```diff
@@ -71,33 +69,26 @@ public final class CASSpinLock: ReadWriteLock {
         static let MASK_READER_BITS          = ~MASK_WRITER_BITS
     }
 
-    private var _state: UnsafeMutablePointer<Int32>
-
-    public init() {
-        _state = UnsafeMutablePointer.allocate(capacity: 1)
-        _state.pointee = 0
-    }
+    private let _state = AtomicInt()
 
-    deinit {
-        _state.deallocate()
-    }
+    public init() {}
```

Comment:
nit: does this need to be here? isn't there a default public init by default?

Reply:
Good catch, I'll try remove that now.